lib/Doctrine/ORM/QueryBuilder.php

```diff
@@ -145,6 +145,13 @@ class QueryBuilder
     protected $lifetime = 0;
 
     /**
+     * The map of query hints.
+     *
+     * @var array
+     */
+    private $_hints = [];
```

Comment:
It would be better to be at after `$joinRootAliases` (the last one of the bundle of private fields)